jena-tdb/src/main/java/org/apache/jena/tdb/store/GraphTDB.java

```diff
@@ -171,7 +171,10 @@ protected final int graphBaseSize() {
 
     @Override
     public void clear() {
-        getDatasetGraphTDB().deleteAny(getGraphName(), Node.ANY, Node.ANY, Node.ANY) ;
+        // Logically, this is "super.clear()" except the default implementation
+        // is a loop that materializes nodes. We want to call the dataset directly 
+        // so that nodes don't get materialized, just deleted from indexes.
```

Comment:
@afs Just for my education, what do you mean by "materialized" here?

Reply:
Loaded from the node table into nodes in the cache I think?

Reply:
Yes.